napi.h

```diff
@@ -201,6 +203,12 @@ namespace Napi {
     Value RunScript(const std::string& utf8script);
     Value RunScript(String script);
 
+    template <typename Hook>
+    CleanupHook<Hook> AddCleanupHook(Hook hook);
+
+    template <typename Hook, typename Arg>
+    CleanupHook<Hook, Arg> AddCleanupHook(Hook hook, Arg* arg);
+
```

Comment:
I think these need to be guarded by NAPI_VERSION > 3 as the cleanup hooks were only added in version 3 as per: https://nodejs.org/api/n-api.html#n_api_napi_add_env_cleanup_hook

Reply:
I thought about this, but we dropped support for node 8, and Node-API version is present in 10+. Do we still need a guard?

Reply:
I think so, people can still ask for a particular Node-API version and I think we should respect that.

Reply:
Addressed.

Reply:
Note I did not add guards to the tests -- we will always test Node-API 3+ right?

Reply:
In terms of guarding the tests its only a 1 line addition here: https://github.com/nodejs/node-addon-api/blob/6697c51d1d08ced8d1c1c40715ceea002613fa6b/test/index.js#L84 so I think we might as well.

Reply:
Thinking more I guess there is more to it than that. You also need to guard the compilation but that is pretty easy too. Unlikely anybody will test with napiVersion <3 but for consistency I'd still prefer if we guard since it's not a lot of effort.

Reply:
Addressed.